src/json.cpp

```diff
@@ -584,6 +584,37 @@ bool JSON::extractstringvalue(const string &json, const string &name, string *va
     return true;
 }
 
+bool JSON::isNumericError(error &e)
+{
+    const char* ptr = pos;
+    if (*ptr == ',')
+    {
+        ptr++;
+    }
```

Comment:
as above, this function is almost the same as isnumeric() and getint(), probably we can just use those in the same way the commands did prevously rather than adding overlapping functionality?

Reply:
There are 2 main differences between `isnumeric` and `inNumericError` is that the later won't return true for positive numbers (errors are always negative or 0, but there are commands that return a positive value on success), and also it returns the `error` by reference by consuming the chars from the JSON in case of negative number (where `isnumeric` + `getint` would have consumed the chars from the JSON without being an error).
I think the difference worth the addition of the `JSON::isNumericError()` (or simply embed the code in the new `checkError()`)